src/System.Reflection.Metadata/tests/PortableExecutable/DebugDirectoryTests.cs

```diff
@@ -153,10 +153,12 @@ private void ValidateEmbeddedPortablePdb(PEReader reader)
             Assert.Equal(DebugDirectoryEntryType.Reproducible, entries[1].Type);
             Assert.Equal(DebugDirectoryEntryType.EmbeddedPortablePdb, entries[2].Type);
 
-            var provider = reader.ReadEmbeddedPortablePdbDebugDirectoryData(entries[2]);
-            var pdbReader = provider.GetMetadataReader();
-            var document = pdbReader.GetDocument(pdbReader.Documents.First());
-            Assert.Equal(@"C:\Documents.cs", pdbReader.GetString(document.Name));
+            using (var provider = reader.ReadEmbeddedPortablePdbDebugDirectoryData(entries[2]))
+            {
+                var pdbReader = provider.GetMetadataReader();
```

Comment:
Yes, we could do that. The MetadataProvider is IDisposable, so I would expect the consumer to use disposable pattern when consuming it and thus keep the provider alive until it's not needed and then dispose it. We also have API MetadataReader.GetBlobReader that returns BlobReader that also doesn't hold the MetadataReader (or anything that creates it) alive.